lib/osf-components/addon/components/registries/finalize-registration-modal/manager/component.ts

```diff
@@ -0,0 +1,53 @@
+import { tagName } from '@ember-decorators/component';
+import { action, computed } from '@ember-decorators/object';
+import Component from '@ember/component';
+import { assert } from '@ember/debug';
+import { layout } from 'ember-osf-web/decorators/component';
+import RegistrationModel from 'ember-osf-web/models/registration';
+import defaultTo from 'ember-osf-web/utils/default-to';
+import template from './template';
+
+export interface FinalizeRegistrationModalManager {
+    registration: RegistrationModel;
+    isOpen: boolean;
+    renderInPlace: boolean;
+    hasEmbargoEndDate: boolean;
+    submitRegistration: () => void;
+    setEmbargoEndDate: (embargoEndDate: Date | null) => void;
+    setCreateDoi: (createDoi: boolean) => void;
+}
+
+@layout(template)
+@tagName('')
+export default class FinalizeRegistrationModalManagerComponent extends Component {
```

Comment:
```suggestion
export default class FinalizeRegistrationModalManagerComponent extends Component
    implements FinalizeRegistrationModalManager {
```
(this will reveal a typo :wink:)

Reply:
The interface is the shape of the `yield`ed hash, not the component's backing class. But in this case, it's both, so maybe that's good

Reply:
Good point. It *probably* make sense, though, to have the properties in the yielded hash match properties in the backing class (seems confusing otherwise). But also, if we had typed templates...

Reply:
It would be cool if you could just denote methods/properties in the backing class as public and they would "automatically" get yielded.

Reply:
Hmm a step in that direction (to get *some* typechecking) could be moving the yielded-hash definition into a computed (would be cleaner with `@tracked`):
```ts
@computed('all', 'the', 'things')
get publicInterface(): MyManager {
  const { all, the, things } = this;
  return { all, the, things };
}
```
```hbs
{{yield this.publicInterface}}
```
(just thinking about how to refine the pattern -- not making a request for this PR)